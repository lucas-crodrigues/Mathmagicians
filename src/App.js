import Header from './components/Header';
import Calculator from './components/Calculator';
import Footer from './components/Footer';
import './Page.css';

const App = () => (
  <div className="app">
    <Header />
    <h3
      style={{
        margin: '0 3vw',
        fontSize: '30px',
      }}
    >
      Let&apos;s do some math!
    </h3>
    <div
      style={{
        display: 'flex',
        flexDirection: 'row-reverse',
      }}
    >
      <div
        style={{
          padding: '0 5vw 5vh 0',
          height: '75vh',
          width: '65vw',
        }}
      >
        <Calculator />
      </div>
    </div>
    <Footer />
  </div>
);

export default App;
