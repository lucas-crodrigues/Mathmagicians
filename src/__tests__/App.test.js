import { BrowserRouter } from 'react-router-dom';
import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import App from '../App';

describe('Calculator page test suites', () => {
  test('Calculator page snapshot', () => {
    const renderedComponent = renderer.create(<BrowserRouter><App /></BrowserRouter>).toJSON();
    expect(renderedComponent).toMatchSnapshot();
  });

  test('Check calculator page title', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
    const calculatorHeading = screen.getByRole('heading', { name: /let's do some math!/i });
    expect(calculatorHeading).toHaveTextContent('Let\'s do some math!');
  });

  test('Check calculator presence', () => {
    render(<BrowserRouter><App /></BrowserRouter>);
    const buttonAC = screen.getByRole('button', { name: /ac/i });
    expect(buttonAC).toBeInTheDocument();
  });
});
