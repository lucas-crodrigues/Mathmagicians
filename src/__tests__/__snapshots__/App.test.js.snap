// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Calculator page test suites Calculator page snapshot 1`] = `
<div
  className="app"
>
  <header
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "lightgray",
        "borderBottom": "1px solid gray",
        "display": "flex",
        "justifyContent": "space-between",
        "padding": "4vh 2vw",
      }
    }
  >
    <h1
      style={
        Object {
          "fontSize": "40px",
        }
      }
    >
      Math Magicians
    </h1>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
          "justifyContent": "space-around",
          "width": "25vw",
        }
      }
    >
      <nav
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
      >
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </nav>
      <div
        style={
          Object {
            "borderLeft": "1px solid gray",
            "height": "5vh",
          }
        }
      />
      <nav
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
      >
        <a
          href="/calculator"
          onClick={[Function]}
        >
          Calculator
        </a>
      </nav>
      <div
        style={
          Object {
            "borderLeft": "1px solid gray",
            "height": "5vh",
          }
        }
      />
      <nav
        style={
          Object {
            "cursor": "pointer",
            "fontSize": "20px",
          }
        }
      >
        <a
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </nav>
    </div>
  </header>
  <h3
    style={
      Object {
        "fontSize": "30px",
        "margin": "0 3vw",
      }
    }
  >
    Let's do some math!
  </h3>
  <div
    style={
      Object {
        "display": "flex",
        "flexDirection": "row-reverse",
      }
    }
  >
    <div
      style={
        Object {
          "height": "75vh",
          "padding": "0 5vw 5vh 0",
          "width": "65vw",
        }
      }
    >
      <section
        className="calculator"
      >
        <div
          className="output"
          data-testid="output"
        >
          0
        </div>
        <div
          className="buttons"
        >
          <button
            className="bt_clear"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
          <button
            className="bt_sign"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
          <button
            className="bt_percentage"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
          <button
            className="bt_division right_column"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
          <button
            className="bt_7"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
          <button
            className="bt_8"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
          <button
            className="bt_9"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
          <button
            className="bt_times right_column"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
          <button
            className="bt_4"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
          <button
            className="bt_5"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
          <button
            className="bt_6"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
          <button
            className="bt_minus right_column"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
          <button
            className="bt_1"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
          <button
            className="bt_2"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
          <button
            className="bt_3"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
          <button
            className="bt_plus right_column"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
          <button
            className="bt_0"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
          <button
            className="bt_dot"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
          <button
            className="bt_equal right_column"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </div>
      </section>
    </div>
  </div>
  <footer
    style={
      Object {
        "backgroundColor": "gray",
        "borderTop": "1px solid black",
        "height": "25px",
      }
    }
  />
</div>
`;
